File: core/src/com/bosowski/oslark/enums/State.java
package com.bosowski.oslark.enums;

public enum State {
    IDLE("idle"), MOVE("move"), ATTACK("attack"), DIE("die"), HIT("hit"), INVALID("invalid");

    public final String name;

    State(String name){
        this.name = name;
    }

    public static State getState(String name){
        switch (name){
            case "idle":
                return IDLE;
            case "move":
                return MOVE;
            case "attack":
                return ATTACK;
            case "die":
                return DIE;
            case "hit":
                return HIT;
            default:
                return INVALID;
        }
    }
}

File: OslarkGame/core/src/com/bosowski/oslark/enums/State.kt
package com.bosowski.oslark.enums

enum class State(val word: String) {
  IDLE("idle"), MOVE("move"), ATTACK("attack"), DIE("die"), HURT("hurt"), INVALID("invalid");


  companion object {

    fun getState(name: String): State {
      return when (name) {
        "idle" -> IDLE
        "move" -> MOVE
        "attack" -> ATTACK
        "die" -> DIE
        "hurt" -> HURT
        else -> INVALID
      }
    }
  }
}
